src/figuresClasses.ts

```diff
@@ -1,19 +1,69 @@
-export interface Figure {
+type ShapeOfFigure = 'triangle' | 'circle' | 'rectangle';
+type ColorOfFigure = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: ShapeOfFigure;
+  color: ColorOfFigure;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: ShapeOfFigure = 'triangle';
 
-}
+  constructor(
+    public color: ColorOfFigure,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (Math.min(this.a, this.b, this.c) <= 0
+    || [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[2]
+    >= [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[0]
+    + [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[1]) {
+      throw new Error('your error message');
```

Comment:
```suggestion
    || [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[2]
    >= [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[0]
    + [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[1]) {
      throw new Error('your error message');
```
you can use the comparison operator because you increase algorithm complexity without necessary and also rewrite this block of code in a separate function